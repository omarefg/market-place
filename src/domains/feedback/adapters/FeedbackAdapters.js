import React, { createContext, useState } from "react";

export const FeedbackContext = createContext();

export function FeedbackProvider(props) {
  const {
    handlersBuilder,
    components,
    useCases,
    children,
    providerMetadata,
  } = props;

  const [textModalContent, setTextModalContent] = useState("");

  const { portal } = providerMetadata;

  const handlers = handlersBuilder({
    stateSetters: {
      setTextModalContent,
    },
    useCases,
  });

  const { handleCloseTextModal } = handlers;

  const { TextModal } = components;

  const value = {
    handlers,
    components,
    state: {
      textModalContent,
    },
    metadata: {
      portal,
    },
  };

  return (
    <FeedbackContext.Provider value={value}>
      {children}
      <TextModal
        open={Boolean(textModalContent)}
        text={textModalContent}
        portal={portal}
        onClose={handleCloseTextModal}
      />
    </FeedbackContext.Provider>
  );
}
