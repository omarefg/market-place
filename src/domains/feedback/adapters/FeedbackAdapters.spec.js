import React from "react";
import { render } from "@testing-library/react";
import { FeedbackProvider } from "./FeedbackAdapters";

const defaultProps = {
  handlersBuilder: () => ({}),
  components: {
    TextModal: () => <div />,
  },
  useCases: {},
  children: <div />,
  providerMetadata: { portal: document.createElement("div") },
};

const wrapper = (props = {}) => {
  const finalProps = { ...defaultProps, ...props };
  return render(<FeedbackProvider {...finalProps} />);
};

describe("@domains/feedback/adapters/FeedbackProvider", () => {
  it("Given a valid call it must render without crashing", () => {
    //Arrange
    const { container } = wrapper();
    //Assert
    expect(container).toMatchSnapshot();
  });

  it("Given a valid call it must render two nodes, children and TextModal", () => {
    //Arrange
    const { container } = wrapper();
    //Assert
    expect(container.childNodes.length).toBe(2);
  });
});
