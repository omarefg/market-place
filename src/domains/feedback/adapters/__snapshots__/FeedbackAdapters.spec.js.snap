// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/feedback/adapters/FeedbackProvider Given a valid call it must render without crashing 1`] = `
<div>
  <div />
  <div />
</div>
`;
